### DISCUSSION TITLE
best diagramming output format to use with asciidoctor? Which kind of dot is used internally?

### DISCUSSION THREAD

**Discussion by aisbergde** [schemacrawler/SchemaCrawler#448] [Q&A] [answered]
Background of the question: I want to use Asciidoc with Antora for database documentation. Asciidoc can use diagram definitions in different formats. Instead of using SC to create only the finished diagrams, I too would like to create and use the plain text diagram definitions (in dot?). These definitions could be integrated into database documentation as not rendered definition, and they can be rendered later into different formats.

from the documentation at https://www.schemacrawler.com/diagramming.html

>SchemaCrawler generates database diagrams using **Graphviz** in any of the output formats supported by Graphviz.

**any** of these formats: http://www.graphviz.org/doc/info/output.html

And on the other hand, https://asciidoctor.org/docs/asciidoctor-diagram/

>Asciidoctor Diagram is a set of Asciidoctor extensions that enable you to add diagrams, which you describe using plain text, to your AsciiDoc document.

>The extensions supports the AsciiToSVG, BlockDiag (BlockDiag, SeqDiag, ActDiag, NwDiag), Bytefield-SVG, Ditaa, dpic, Erd, Gnuplot, **GraphViz**, Mermaid, Msc, Nomnoml, Pikchr, PlantUML, Shaape, State Machine Cat, SvgBob, Symbolator, Syntrax, UMLet, Vega, Vega-Lite and WaveDrom syntax.

So both SchemaCrawler and asciidoctor-diagram can use one or more common formats: SC can export to a format from which asciidoc can then create the final diagrams. Probably this format is "dot"?

But there seem to be different variants of "dot":

- canon
- dot
- gv
- xdot
- xdot1.2
- xdot1.4

**Which format is used by SchemaCrawler internally? Which output format is the most lossless?**

**Comment by sualeh** [comment, marked as answer]
Use the undocumented SchemaCrawler output format"scdot", which produces Graphviz dot format.

**Comment by aisbergde** [comment]
I can generate these graphviz files, I can render them in VSC using the graphviz plugin, this works fine.

On the other hand, it is probably better to use the SVGs created by SC: you can look at them right away. And also in Antora or with asciidoc they would have to be converted first :-)
